### PR TITLE
Standardize project/event/org language on "organization"

### PR DESCRIPTION
**Pull request by lachlanjc**
## Summary of the problem

We've been using a variety of terms for what are "events" in our code publicly: event, project, nonprofit, organization. Talked to @dwahbe & decided to standardize on "organization"—teenagers want to be part of organizations, not on projects.

## Describe your changes

Transparency:
CleanShot 2024-05-15 at 21 34 28@2x

Homepage:
CleanShot 2024-05-15 at 21 34 50@2x

I've also updated links to hackclub.com/hcb to https://hackclub.com/fiscal-sponsorship/ since those were out of date.

### CODE REVIEW

File: app/helpers/events_helper.rb [hunk 1]
```diff
@@ -91,6 +91,6 @@ def render_audit_log_value(field, value, color:)
   end
 
   def show_org_switcher?
-    Flipper.enabled?(:org_switcher_2024_01_31, current_user) && current_user.events.not_hidden.count > 1
+    signed_in? && Flipper.enabled?(:org_switcher_2024_01_31, current_user) && current_user.events.not_hidden.count > 1
```
Comment (sarthaktexas) [inline app/helpers/events_helper.rb]: Do we need this if the org switcher is only shown on pages that are shown to signed in people? I think the flipper flag should take care of this, no?

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -89,9 +89,9 @@
     <% end %>
     <% if @event&.is_public && !organizer_signed_in? && !@no_app_shell && !@no_transparency_header && !@use_user_nav %>
       <div class="w-100 px3 bg-white transparency-banner" data-behavior="hide_iframe">
-        <p class="h5 medium center mt1 mb1">
-          This HCB project has been made public so you can see all its finances.
-          Hey, <%= link_to "what’s HCB?", "https://hackclub.com/hcb/", target: "_blank", class: "primary" %>
+        <p class="h5 medium center text-balance mt1 mb1">
```
Comment (sarthaktexas) [inline app/views/layouts/application.html.erb]:
```suggestion
        <p class="h5 font-medium text-center text-balance mt1 mb1">
```

File: app/views/static_pages/index.html.erb [hunk 3]
```diff
@@ -94,8 +94,8 @@
       </ul>
     <% end %>
     <% if @events.empty? && @invites.empty? && !current_user.admin? %>
-      <%= blankslate "No projects associated with your account yet." %>
-      <a href="https://hackclub.com/hcb/apply/?userEmail=<%= current_user.email %>&firstName=<%= current_user.first_name %>&lastName=<%= current_user.last_name %>&returningUser=No,%20first%20time!&userPhone=<%= current_user.phone_number %>" class="btn bg-success">
+      <%= blankslate "No organizations associated with your account yet." %>
+      <a href="https://hackclub.com/fiscal-sponsorship/apply/?userEmail=<%= current_user.email %>&firstName=<%= current_user.first_name %>&lastName=<%= current_user.last_name %>&returningUser=No,%20first%20time!&userPhone=<%= current_user.phone_number %>" class="btn bg-success">
```
Comment (sarthaktexas) [inline app/views/static_pages/index.html.erb]: We don't have a `returning user field` anymore on the application. We should remove from here too.
```suggestion
      <a href="https://hackclub.com/fiscal-sponsorship/apply/?userEmail=<%= current_user.email %>&firstName=<%= current_user.first_name %>&lastName=<%= current_user.last_name %>&userPhone=<%= current_user.phone_number %>" class="btn bg-success">
```

**Review by sarthaktexas — COMMENTED** [review]
This is an initial review

File: app/helpers/events_helper.rb (hunk 1 above)
Comment (lachlanjc) [reply app/helpers/events_helper.rb]: I was getting an error in dev trying to call `current_user` here from transparency mode, & this solved it. We def don't want the org switcher shown in transparency

**Review by garyhtou — APPROVED** [review]
Thanks for doing this @lachlanjc! It's very much needed

**Pull request merged**